packages/web-components/fast-foundation/src/carousel/carousel.ts

```diff
@@ -0,0 +1,411 @@
+    @attr({ attribute: "activeslideid" })
+    public activeSlideId: string;
+    public activeSlideIdChanged(): void {
+        this.activeSlideIndex = this.slideIds.indexOf(this.activeSlideId);
+    }
+
+    @attr({ attribute: "aria-labelledby" })
+    public arialabelledby: string;
+
+    @attr({ attribute: "aria-label" })
+    public arialabel: string;
+
+    @observable
+    public focused: boolean = false;
+    public focusedChanged() {
+        if (this.focused) {
+            this.notabfocus = false;
+        } else {
+            this.notabfocus = true;
+        }
+    }
+
+    /**
+     * Whether or not to focus the tab on change
+     * @public
+     * HTML Attribute: notabfocus
+     */
+    @observable
+    public notabfocus: boolean = true;
+    public nottabfocusChanged(): void {}
+
+    public setComponent(): void {
+        this.activeid = this.tabIds[this.activeTabIndex] as string;
+        this.change();
+        this.setTabs();
+        this.handleActiveIndicatorPosition();
+        this.setTabPanels();
+        if (!this.notabfocus) {
+            this.focusTab();
+        }
+        this.change();
+    }
+
+    public activeindicator = false;
+
+    @observable
+    public activeSlideIndex: number = 0;
+    public basicContent: HTMLDivElement;
+    public carousel: HTMLDivElement;
+    public previousFlipperDefault: HTMLElement;
+    public previousFlipperSlottedItem: HTMLElement[];
+    public nextFlipperDefault: HTMLElement;
+    public nextFlipperSlottedItem: HTMLElement[];
+    public rotationControl: HTMLElement;
+    public rotationControlItem: HTMLElement[];
+    public tablistRef: HTMLElement;
+    public tabPanelsContainerRef: HTMLElement;
+
+    @observable
+    public items: HTMLElement[];
+    public itemsChanged(): void {
+        if (this.items.length && this.basicPattern) {
+            this.generateSlideIds();
+
+            //sethdonohue - if activeSlideId attribute was set by implementation then we need to sync the activeSlideIndex for incrementing to work
+            if (this.activeSlideId) {
+                this.activeSlideIndex = this.slideIds.indexOf(this.activeSlideId);
+            } else {
+                this.activeSlideId = this.slideIds[this.activeSlideIndex] as string;
+            }
+
+            this.items.forEach((item: HTMLElement, index: number) => {
+                if (index === this.activeSlideIndex) {
+                    item.classList.add("active-slide");
+                    item.removeAttribute("hidden");
+                } else {
+                    item.setAttribute("hidden", "");
+                }
+                if (
+                    !item.getAttribute("aria-label") ||
+                    !item.getAttribute("aria-labelledby")
+                ) {
+                    item.setAttribute(
+                        "aria-label",
+                        `${index + 1} of ${this.items.length}`
+                    );
+                }
+
+                item.setAttribute(
```

Comment:
You only need to invoke `setAttribute` if the id isn't already there - maybe update to not invoke it if that case is met.